### DISCUSSION TITLE
[question] [typescript] Problem with global types

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [oven-sh/bun#4750] [Q&A]
`Can not find name 'Bun'` TypeScript error.

Tried installing bun-types with `bun add bun-types`.

Vanilla bun project, just a `main.ts`.

Latest VSCode Windows 11 Home Ubuntu WSL terminal, bun x1, 

Everything seems ok at runtime.

Congratulations for x1 release! This runtime is promising!

**Comment by pgmichael** [comment]
Yeah, looks like nothing was handled out of the box for me even with the extension installed.

I just installed the types manually (`bun add bun-types`) and created a `tsconfig.json` file at the root of my project containing the following code:

```json
{
  "compilerOptions": {
    "types": ["bun-types"],
  }
}
```

Hope this helps :)

**Comment by angelhdzmultimedia** [comment]
> Yeah, looks like nothing was handled out of the box for me even with the extension installed.
> 
> I just installed the types manually (`bun add bun-types`) and created a `tsconfig.json` file at the root of my project containing the following code:
> 
> ```json
> {
>   "compilerOptions": {
>     "types": ["bun-types"],
>   }
> }
> ```
> 
> Hope this helps :)

Thank you so much for your time and the info.

I did exactly that and reloaded the TS language server.
What now? 🥲

![image](https://github.com/oven-sh/bun/assets/10191085/6560abc6-892a-4856-88c6-02aff7b3738c)

**Comment by angelhdzmultimedia** [comment]
### Solved
I installed `bun-types` and added it to `compilerOptions.types` array and it didn't work... but then I executed `bun install` and now it works.

![image](https://github.com/oven-sh/bun/assets/10191085/a42f5046-dc24-4f8a-84b1-a77375142338)

![ElmoSmygbaGIF](https://github.com/oven-sh/bun/assets/10191085/511b87a0-f8fa-48d2-9f6a-d508cb707abd)

**Comment by angelhdzmultimedia** [comment]
What caught my attention about bun, wasn't speed (everyone is obsessed with authoring/compiling/run time speed these days), was the "out of the box" characteristic, but having to install the types separately gave me  a bad impression.

Now wondering "what else I have to add now?" I also had to activate strict mode, I expected it to be enabled by default...

**Comment by souporserious** [comment]
I was able to do `bun add @types/bun` and it fixed the type error in VS Code without any additional configuration.
